### PR TITLE
Add api to get cluster id from usage state head

### PR DESCRIPTION
**Pull request by brycehuang30**
## Why are these changes needed?

Added an API to expose cluster id. This will be used by third-party to get the cluster_id of the Ray cluster.

## Result

Manually tested that the cluster id can be returned from dashboard (I set the dashboard at port 8888)

Screenshot 2024-05-03 at 4 48 09 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — COMMENTED** [review]
LGTM.

Can you add a test?

File: dashboard/BUILD [hunk 1]
```diff
@@ -112,3 +113,11 @@ py_test(
     tags = ["team:data"],
     deps = [":conftest"],
 )
+
+py_test(
+    name = "test_usage_stats_head",
+    size = "small",
+    srcs = ["tests/test_usage_stats_head.py"],
+    tags = ["team:core"],
```
Comment (brycehuang30) [inline dashboard/BUILD:121]: I assign this to core team given that the usage_stats_head is created by core team.

File: dashboard/tests/test_usage_stats_head.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    assert ray._private.utils.check_dashboard_dependencies_installed()
+
+    ray.init()
+
+    data = requests.get("http://localhost:8265/cluster_id").json()
+
+    assert data["result"] is True
+    assert data["msg"] == "Fetched cluster id"
+    gcs_client = GcsClient(address=ray.get_runtime_context().gcs_address)
+    assert data["data"]["clusterId"] == gcs_client.cluster_id.hex()
```
Comment (brycehuang30) [inline dashboard/tests/test_usage_stats_head.py:19]: Not sure whether this is the best practice to check cluster id internally

**Comment by brycehuang30** [comment]
Thanks @jjyao for reviewing. I added a test for this API, could you take another look?

File: dashboard/tests/test_usage_stats_head.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import sys
+import ray
+import requests
+import pytest
+from ray._raylet import GcsClient
+
+
+@pytest.mark.asyncio
```
Comment (jjyao) [inline dashboard/tests/test_usage_stats_head.py:8]: You can add the test to the existing `test_usage_stats.py`. Check `test_usage_stats_enabled_endpoint`

Comment (brycehuang30) [reply dashboard/tests/test_usage_stats_head.py:8]: Got it! Let me move the test there

**Review by jjyao — APPROVED** [review]

**Pull request merged**